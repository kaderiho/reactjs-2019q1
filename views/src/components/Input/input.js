"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var Input = function Input(props) {
  var label = props.label,
      input = (0, _objectWithoutProperties2.default)(props, ["label"]);
  return _react.default.createElement("div", {
    className: "inputHolder"
  }, props.label && _react.default.createElement("label", {
    className: "inputHolder-label",
    htmlFor: props.name
  }, props.label), _react.default.createElement("input", (0, _extends2.default)({
    className: "inputHolder-input"
  }, input)));
};

var _default = Input;
exports.default = _default;