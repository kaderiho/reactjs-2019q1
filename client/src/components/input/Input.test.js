import React from 'react';
import { shallow, mount } from 'enzyme';
import Input from './Input';

const inputChangeFn = jest.fn();

/**
 * Tests:
 * 
 * - Snapshot match
 * - onChange handler call
 * - Label existing
 */
describe('<Input /> component', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<Input />);

        expect(wrapper).toMatchSnapshot();
    });

    it('onChange handler shoud be called', () => {
        const wrapper = mount(<Input onChange={inputChangeFn} />);

        wrapper.find('input').simulate('change');

        expect(inputChangeFn).toHaveBeenCalled();
    });

    it('Label should exist in case of passed label text', () => {
        const wrapper = mount(<Input label="Some label" />);

        expect(wrapper.exists('.inputHolder-label')).toEqual(true);
    });
});