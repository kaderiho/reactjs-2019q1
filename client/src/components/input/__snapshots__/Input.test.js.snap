// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component Should render correctly 1`] = `
<div
  className="inputHolder"
>
  <input
    className="inputHolder-input"
  />
</div>
`;
